internal/gcsx/client_readers/gcs_reader.go

```diff
@@ -116,6 +116,7 @@ func (gr *GCSReader) ReadAt(ctx context.Context, p []byte, offset int64) (gcsx.R
 	}
 	defer func() {
 		gr.updateExpectedOffset(offset + int64(readerResponse.Size))
+		gr.totalReadBytes += uint64(readerResponse.Size)
```

Comment:
Good find. 

Is it possible to write an e2e test which basically validates number of GCS calls for a sequential read?

Reply:
I'll handle that as a separate task(will a create bug), as it seems to require significant effort. Until then, we can consider an FIO test with a 200MiB file, using both 1MiB and 4KiB block sizes, a good verification test.